applications/TopologyOptimizationApplication/custom_elements/small_displacement_simp_element.cpp

```diff
@@ -0,0 +1,302 @@
+{
+    return Kratos::make_intrusive<SmallDisplacementSIMPElement>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Create( IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties ) const
+{
+    return Kratos::make_intrusive<SmallDisplacementSIMPElement>( NewId, pGeom, pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+SmallDisplacementSIMPElement::~SmallDisplacementSIMPElement()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Clone (
+    IndexType NewId,
+    NodesArrayType const& rThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    SmallDisplacementSIMPElement::Pointer p_new_elem = Kratos::make_intrusive<SmallDisplacementSIMPElement>(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+    p_new_elem->SetData(this->GetData());
+    p_new_elem->Set(Flags(*this));
+
+    // Currently selected integration methods
+    p_new_elem->SetIntegrationMethod(BaseType::mThisIntegrationMethod);
+
+    // The vector containing the constitutive laws
+    p_new_elem->SetConstitutiveLawVector(BaseType::mConstitutiveLawVector);
+
+    return p_new_elem;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+
+// =============================================================================================================================================
+// STARTING / ENDING METHODS
+// =============================================================================================================================================
+
+//************************************************************************************
+//************************************************************************************
+
+	
+void SmallDisplacementSIMPElement::CalculateOnIntegrationPoints(const Variable<double>& rVariable, std::vector<double>& rValues,
+        const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Additional part for post-processing of the topology optimized model part
+    if (rVariable == X_PHYS)
+		//CalculateOnIntegrationPoints(rVariable, rValues, rCurrentProcessInfo);
+        ;
+    // From original SmallDisplacementElement
+    else if (rVariable == VON_MISES_STRESS)
```

Comment:
If unused removed, otherwise will compute nothing (Will not call the base class method)